client/components/account-balances/index.tsx

```diff
@@ -115,7 +115,7 @@ const AccountBalances: React.FC = () => {
 			currencyCode: overview.currency,
 			availableFunds: overview.available?.amount ?? 0,
 			pendingFunds: overview.pending?.amount ?? 0,
-			delayDays: account.deposits_schedule.delay_days,
+			delayDays: account?.deposits_schedule.delay_days,
```

Comment:
tl;dr; this comment is just an FYI. Not suggesting any change.

Tempted to suggest a default value so in accordance of the [loading state](https://github.com/Automattic/woocommerce-payments/blob/a638d5599d8a09d14f5826a0e6af6492830a3374/client/components/account-balances/index.tsx#L79).

```suggestion
			delayDays: account?.deposits_schedule.delay_days ?? 0,
```

However, in case of `account` is null (which by itself is impossible to pass the `if ( isLoading )` check) `account?.deposits_schedule.delay_days` being undefined looks more correct:
![Screenshot 2023-12-18 at 02 19 23](https://github.com/Automattic/woocommerce-payments/assets/73803630/f9e50516-a92b-4675-9e67-92f27e4e1e28)

than if we set `delayDays` to be 0:
![Screenshot 2023-12-18 at 02 19 37](https://github.com/Automattic/woocommerce-payments/assets/73803630/b6d29d67-506f-4f50-bcfd-2e90d787b1b3)

Also I wondered why TS does not complain that `delayDays` can be undefined although [delayDays' type is defined as a `number`](https://github.com/Automattic/woocommerce-payments/blob/a638d5599d8a09d14f5826a0e6af6492830a3374/client/components/account-balances/index.tsx#L41). Maybe because [here](https://github.com/Automattic/woocommerce-payments/blob/a638d5599d8a09d14f5826a0e6af6492830a3374/client/components/account-balances/index.tsx#L111) we don't strictly define the type of `depositCurrencyTabs` to be `BalanceTabProps[]`.

Reply:
Another thing we could do to tidy this up – we can define and return 2 distinct types: one while loading and one loaded. E.g.

```ts
export interface OverviewsResponse {
  overviews: OverviewsLoading | Overviews;
  isLoading: boolean;
}

export interface OverviewsLoading {
  account: null;
  currencies: Overview[];
  isLoading: true;
}

export interface Overviews {
  account: Account;
  currencies: Overview[];
  isLoading: false;
}

export function isOverviewsLoading(response: OverviewsResponse): response is OverviewsLoading {
  return response.isLoading;
}
```

Then for the `isLoading` check, we can use the type assertion function `isOverviewsLoading`. If that returns false, all following logic will be confident that `account: Account`, no need for `??` to account for possible `account: null`.

Reply:
I've used your solution in eaa6dce as a good solution for now 👍

Reply:
> we can define and return 2 distinct types: one while loading and one loaded

Ow wow, that's actually neat.